### DISCUSSION TITLE
ESPHome

### DISCUSSION THREAD

**Discussion by AquaMCU** [timmbogner/Farm-Data-Relay-System#55] [Ideas]
Would it be possible to merge this project with ESPHome? 

https://esphome.io/

What I would think we could do:
This integration would enable an easy installation of the software ( the build stack is in a super docker container ) and would provide a lot of features out of the box

- Easy deployment using docker
- OTA Update
- Access to a ton of integrations
- Integration to Home Assistant
-  Vivit community to support this project as well
-  Lack of Lora integration

In the bug tacker, there is a vivit discussion on how to integrate lora, so there is a need for a software like this

Looking forward discussing this

Oliver

**Comment by timmbogner** [comment]
FDRS is designed with modularity in mind, so projects like ESPHome/HA should have no issues integrating with FDRS. They can communicate with a gateway via serial, MQTT, or even ESP-NOW in order to retrieve sensor values.
As for merging the two: No, FDRS will remain an independent tool for light-weight communication. No frills, it just moves data around.

**Comment by AquaMCU** [comment]
Understood and respected. Do you plan on creating a docker like build environment? Installing the IDE is a pain for most end customers. 

Something like this: https://hub.docker.com/r/arduino/arduino-cli

Oliver

**Reply by timmbogner** [reply]
It might end up similar! My dream is an actual app with GUI and GIS that lets you drag and drop your gateways/sensors on a map, then walks you through configuring and flashing each device. Not sure if esphome/HA have anything like that or not. Underneath the GUI there would probably be something like what you're looking for.
I also used WLED's web-usb-serial interface for the first time this spring and _loved it_., so it could end up more like that too. I don't know much about higher-level software, hence it's a dream for now.
